### DISCUSSION TITLE
Single layer app with DbMigrator project

### DISCUSSION THREAD

**Discussion by StefanIvovic** [abpframework/abp#12872] [Q&A]
Hi, i really like using DbMigrator in layered web app solution and i was trying to create it in Single layer web app but unfortunately i am getting exception witch i cant solve. 

exception is thrown in DbMigratorHostedService.StartAsync() application.Initialize();
The requested service 'Microsoft.AspNetCore.Hosting.IWebHostEnvironment' has not been registered.

using .net 6

**Comment by berkansasmaz** [comment]
Can you try again with the code below?

> After creating your solution, you need to create the database before running the application. We've added a parameter to the application that can be specified to migrate the database and seed the initial data. Open the project's directory (that contains the csproj file) in a command-line terminal and type the following command:

`dotnet run --migrate-database`

**Reference:**
- https://github.com/abpframework/abp/blob/dev/docs/en/Blog-Posts/2022-03-08%20v5_2_Preview/POST.md#database-migrations-for-ef-core

**Reply by StefanIvovic** [reply]
Ok, how would i select production connection string

**Reply by StefanIvovic** [reply]
something like dotnet run --migrate-database ASPNETCORE_ENVIRONMENT="Production"  ?

**Reply by berkansasmaz** [reply]
For `macOS`, you can try the below command. A detailed explanation for `Windows` is available [here](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-6.0#windows---set-environment-variable-for-a-process).

`ASPNETCORE_ENVIRONMENT=Production dotnet run --migrate-database`

Data is seeded for both `development` and `production` by default. If you want to be seeded only in `development` mode, you need to customize the `BookStoreDbMigrationService` class in the `Data `folder according to your needs.
